### PR TITLE
RPC profiling recipe

### PR DESCRIPTION
**Pull request by rohan-varma**
Adding a recipe for using profiler with distributed RPC framework.

Test plan: View the output https://deploy-preview-1068--pytorch-tutorials-preview.netlify.app/recipes/distributed_rpc_profiling.html

### CODE REVIEW

**Comment by rohan-varma** [comment]
cc @mrshenli @jlin27

File: recipes_source/distributed_rpc_profiling.rst [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+            with profiler.profile() as prof:
+                fut1 = rpc.rpc_async(dst_worker_name, torch.add, args=(t1, t2))
+                fut2 = rpc.rpc_async(dst_worker_name, torch.mul, args=(t1, t2))
+                # RPCs must be awaited within profiling scope.
+                fut1.wait()
+                fut2.wait()
+
+            print(prof.key_averages().table())
+
+The aformented code creates 2 RPCs, specifying ``torch.add`` and ``torch.mul``, respectively,
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:122]: do you mean "aformented" -> "aforementioned"

File: recipes_source/distributed_rpc_profiling.rst [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+We also have visibility into the actual operators invoked remotely on worker 1 due RPC.
+We can see operations that took place on ``worker1`` by checking the ``Node ID`` column. For 
+example, we can interpret the row with name ``rpc_async#aten::mul(worker0 -> worker1)#remote_op: mul``
+as a ``mul`` operation taking place on the remote node, as a result of the RPC sent to ``worker1``
+from ``worker0``, specifying ``worker1`` to run the builtin ``mul`` operator on the input tensors.
+Note that names of remote operations are prefixed with the name of the RPC event that resulted
+in them. For example, remote operations corresponding to the ``rpc.rpc_async(dst_worker_name, torch.add, args=(t1, t2))``
+call are prefixed with ``rpc_async#aten::mul(worker0 -> worker1)``.
+
+We can also use the profiler gain insight into user-defined functions that are executed over RPC.
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:160]: gain -> to gain

File: recipes_source/distributed_rpc_profiling.rst [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+      import time
+      time.sleep(1)
+      t1, t2 = random_tensor(), random_tensor()
+      torch.add(t1, t2)
+      torch.mul(t1, t2)
+
+  def worker(rank, world_size):
+      # Above code omitted
+      with profiler.profile() as p:
+          fut = rpc.rpc_async(dst_worker_name, udf_with_ops, args=())
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:176]: we can leave `args=()` out?

File: recipes_source/distributed_rpc_profiling.rst [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+remote operators that have been executed on worker 1 as part of executing this RPC request.
+
+Lastly, we can visualize remote execution using the tracing functionality provided by the profiler.
+Let's add the following code to the above ``worker`` function:
+
+::
+
+    def worker(rank, world_size):
+        # Above code omitted
+        # Will generated trace for above profiling output
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:213]: generated -> generate

File: recipes_source/distributed_rpc_profiling.rst [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+        logger.debug(f"Wrote trace to {trace_file}")
+
+Now, we can load the trace file in Chrome (``chrome://tracing``). We should see output similar to
+the following:
+
+.. image:: ../_static/img/rpc_trace_img.png
+   :scale: 25 %
+
+As we can see, we have traced our RPC requests and can also visualize traces of the remote operations,
+in this case, given in the trace column for ``node_id: 1``.
```
Comment (mrshenli) [inline recipes_source/distributed_rpc_profiling.rst:225]: do you mean trace row?

**Pull request merged**